plugins/module_utils/consul.py

```diff
@@ -211,6 +233,13 @@ def prepare_object(self, existing, obj):
             existing[k] = v
         return existing
 
+    def get_unique_identifier(self):
```

Comment:
This feels a bit like a leaky abstraction. If you consider this example (copied verbatim from below):
```python
        identifier = self.get_unique_identifier()
        url = self.endpoint_url(
            OPERATION_UPDATE, existing.get(camel_case_key(identifier))
        )
```
Here we are calculating the identifier based on `self.params` but take the value from `existing`.

What do you think about writing a function like:
```python
def id_from_obj(obj, camel_case=False):
        def key_func(key):
              return camel_case_key(key) if camel_case else key

        if self.unique_identifiers:
            for identifier in self.unique_identifiers:
                identifier = key_func(identifier)
                if identifier in obj:
                    return obj[identifier]
        raise RuntimeError("No valid identifier found")
```

then you can use `self.id_from_obj(existing, camel_case=True)` or `self.id_from_obj(self.params)` as needed.

I also removed the handling of `self.unique_identifier`, I would simply do:
```
if self.unique_identifier:
   self.unique_identifiers = [self.unique_identifier]
```
in `__init__`.

Does that make sense to you?

Reply:
FWIW I would be fine with removing `unique_identifier` completely and require `unique_identifiers` always to be set.

Reply:
Thank you for helping me. This makes total sense to me. 
For the self.unique_identifiers, I would add it in this PR and remove self.unique_identifier in a future PR. 

If I want to add it as a PR first, then I need to do the following steps ? 
1. new branch and implement unique_identifiers
2. let it merge in main 
4. rebase this branch ? 
Is this more or less the way to do it ?

Reply:
I think if you remove `unique_idenfitier` in a future PR you can just do the other changes here.